R/class-workbook.R

```diff
@@ -5400,6 +5728,281 @@ wbWorkbook <- R6::R6Class(
+            self$worksheets[[sheet]]$extLst <- newExtLst
+          } else if (length(xml_node(extLst, "ext", "x14:conditionalFormattings")) == 0) {
+            # extLst is available, has no conditionalFormattings
+            extLst <- xml_add_child(extLst,
+                                    xml_node(newExtLst, "ext", "x14:conditionalFormattings"))
+            self$worksheets[[sheet]]$extLst <- extLst
+          } else {
+            # extLst is available, has conditionalFormattings
+            extLst <- xml_add_child(extLst,
+                                    xml_node(newExtLst, "ext", "x14:conditionalFormattings", "x14:conditionalFormatting"),
+                                    level = "x14:conditionalFormattings")
+            self$worksheets[[sheet]]$extLst <- extLst
+          }
+
+
+
+          if (is.null(values)) {
+            sprintf(
+              '<cfRule type="dataBar" priority="1"><dataBar showValue="%s">
+                          <cfvo type="min"/><cfvo type="max"/>
+                          <color rgb="%s"/>
+                          </dataBar>
+                          <extLst><ext uri="{B025F937-C7B1-47D3-B67F-A62EFF666E3E}" xmlns:x14="http://schemas.microsoft.com/office/spreadsheetml/2009/9/main"><x14:id>{%s}</x14:id></ext>
+                        </extLst></cfRule>',
+              params$showValue,
+              posColour,
+              guid
+            )
+          } else {
+            sprintf(
+              '<cfRule type="dataBar" priority="1"><dataBar showValue="%s">
+                            <cfvo type="num" val="%s"/><cfvo type="num" val="%s"/>
+                            <color rgb="%s"/>
+                            </dataBar>
+                            <extLst><ext uri="{B025F937-C7B1-47D3-B67F-A62EFF666E3E}" xmlns:x14="http://schemas.microsoft.com/office/spreadsheetml/2009/9/main">
+                            <x14:id>{%s}</x14:id></ext></extLst></cfRule>',
+              params$showValue,
+              values[[1]],
+              values[[2]],
+              posColour,
+              guid
+            )
+          }
+        },
+
+        expression = {
+          sprintf(
+            '<cfRule type="expression" dxfId="%s" priority="1"><formula>%s</formula></cfRule>',
+            dxfId,
+            formula
+          )
+        },
+
+        duplicatedValues = {
+          sprintf(
+            '<cfRule type="duplicateValues" dxfId="%s" priority="1"/>',
+            dxfId
+          )
+        },
+
+        containsText = {
+          sprintf(
+            '<cfRule type="containsText" dxfId="%s" priority="1" operator="containsText" text="%s">
+                        	<formula>NOT(ISERROR(SEARCH("%s", %s)))</formula>
+                       </cfRule>',
+            dxfId,
+            values,
+            values,
+            # is this unlist correct?  Would this not work?
+            # > strsplit(sqref, split = ":")[[1]]
+            unlist(strsplit(sqref, split = ":"))[[1]]
```

Comment:
Agree, looks a bit strange. I felt that this entire code section was a bit fragile when I went over it a little while ago. Minor changes have a tendency to cause broken xlsx files.
This will actively select only the first element. It might be that the input is `A:A` and we only want `A`)

Reply:
Ah, I meant `strsplit(sqref, split = ":")[[1]][1]`, which makes a bit more sense.